1.8/usage/registry.md

```diff
@@ -0,0 +1,167 @@
+1. Configure Docker daemons to trust the self-signed certificates.
+
+Open an SSH connection to a master node in the cluster with credentials forwarding enabled (option `-A`), so that an SSH connection can be made to other nodes in the cluster. Note: this tutorial uses [jq](https://stedolan.github.io/jq/), but you can also carry out the steps using, for example, `grep`.
+
+Apply this workaround for minuteman to all of your agents so that you can push and pull images to and from the registry.
+
+```bash
+$ for i in $(curl -sS master.mesos:5050/slaves | jq '.slaves[] | .hostname' | tr -d '"'); do ssh "$i" -oStrictHostKeyChecking=no "sudo sysctl -w net.netfilter.nf_conntrack_tcp_be_liberal=1"; done
+```
+
+### Create a self signed cert
+
+The first step is to create a self-signed certificate to secure a Docker registry. It’s important that the common name used in the certificate matches the domain name that is used to access the registry.
+
+You can create a self-signed certificate to secure a registry by using OpenSSL. This command creates a certificate that uses a minuteman VIP to access the registry. 
+
+```bash
+$ openssl req -newkey rsa:2048 -nodes -keyout domain.key -x509 -days 365 -out domain.crt -subj "/C=US/ST=NY/L=NYC/O=Mesosphere/CN=registry.marathon.l4lb.thisdcos.directory"
+```
+### Copy the certificate and key to the agents
+
+All of the Docker daemons in your cluster must be configured to trust the self-signed certificate. To configure this, you must distribute the public certificate on every agent node running Docker, and on the masters that will connect to the registry.
+
+The Docker registry itself requires the certificate and the private key. You can embed these in the Docker image by building a new Docker image from `registry:2` and including the certificate files. You can also use a configuration management tool or script to copy these files to every agent in the cluster that the registry service is scheduled on. The files could also come from a file share mounted to all of the nodes, like Azure Files.
+
+You can use the following commands to copy the certificate and keys to all agents in the cluster:
+
+```bash
+$ MESOS_AGENTS=$(curl -sS master.mesos:5050/slaves | jq '.slaves[] | .hostname' | tr -d '"');
+$ for i in $MESOS_AGENTS; do ssh "$i" -oStrictHostKeyChecking=no "sudo mkdir --parent /etc/privateregistry/certs/"; done
+$ for i in $MESOS_AGENTS; do scp -o StrictHostKeyChecking=no ./domain.* "$i":~/; done
+$ for i in $MESOS_AGENTS; do ssh "$i" -oStrictHostKeyChecking=no "sudo mv ./domain.* /etc/privateregistry/certs/"; done
+```
+
+You must configure the Docker daemon on all machines that require registry access to trust the self-signed certificate. You must  create a folder that matches the FQDN in the `certs.d` folder and copy the certificate to the folder using the name `ca.crt`. You can use the following script to create the folders and copy the files into the appropriate locations:
```

Comment:
Use this script to:

* Configure the Docker daemon on all machines that require registry access to trust the self-signed certificate and create a folder that matches the FQDN in the `certs.d` folder.
* Copy the certificate to the folder using the name `ca.crt`.`